library/src/scala/runtime/LazyVals.scala

```diff
@@ -37,6 +38,41 @@ object LazyVals {
 
   /* ------------- Start of public API ------------- */
 
+  /**
+   * Used to indicate the state of a lazy val that is being
+   * evaluated and of which other threads await the result.
+   */
+  final class Waiting:
+    private var done = false
+    
+    /**
+     * Wakes up waiting threads. Called on completion of the evaluation
+     * of lazy val's right-hand side.
+     */
+    def release(): Unit = synchronized {
```

Comment:
`synchronized` is not supported by project Loom (it uses native calls so blocks OS threads), so @sjrd comment about latches is the officially recommended workaround

Reply:
It seems that after introducing CountDownLatch, the synchronized is not required.